src/adapter/src/coord.rs

```diff
@@ -1717,6 +1587,94 @@ impl Coordinator {
+    /// storage collections, without needing to worry about dependency order.
+    async fn bootstrap_storage_collections(&mut self) {
+        let catalog = self.catalog();
+        let source_status_collection_id = catalog
+            .resolve_builtin_storage_collection(&mz_catalog::builtin::MZ_SOURCE_STATUS_HISTORY);
+
+        let source_desc = |source: &Source| {
+            let (data_source, status_collection_id) = match &source.data_source {
+                // Re-announce the source description.
+                DataSourceDesc::Ingestion(ingestion) => {
+                    let ingestion = ingestion.clone().into_inline_connection(catalog.state());
+
+                    (
+                        DataSource::Ingestion(ingestion.clone()),
+                        Some(source_status_collection_id),
+                    )
+                }
+                // Subsources use source statuses.
+                DataSourceDesc::Source => (
+                    DataSource::Other(DataSourceOther::Source),
+                    Some(source_status_collection_id),
+                ),
+                DataSourceDesc::Webhook { .. } => {
+                    (DataSource::Webhook, Some(source_status_collection_id))
+                }
+                DataSourceDesc::Progress => (DataSource::Progress, None),
+                DataSourceDesc::Introspection(introspection) => {
+                    (DataSource::Introspection(*introspection), None)
+                }
+            };
+            CollectionDescription {
+                desc: source.desc.clone(),
+                data_source,
+                since: None,
+                status_collection_id,
+            }
+        };
+
+        let collections: Vec<_> = catalog
+            .entries()
+            .filter_map(|entry| {
+                let id = entry.id();
+                match entry.item() {
+                    CatalogItem::Source(source) => Some((id, source_desc(source))),
+                    CatalogItem::Table(table) => {
+                        let collection_desc = CollectionDescription::from_desc(
+                            table.desc.clone(),
+                            DataSourceOther::TableWrites,
+                        );
+                        Some((id, collection_desc))
+                    }
+                    CatalogItem::MaterializedView(mv) => {
+                        let collection_desc = CollectionDescription::from_desc(
+                            mv.desc.clone(),
+                            DataSourceOther::Compute,
+                        );
+                        Some((id, collection_desc))
+                    }
+                    _ => None,
+                }
+            })
+            .collect();
+
+        self.controller
+            .storage
+            .migrate_collections(collections.clone())
+            .await
+            .unwrap_or_terminate("cannot fail to migrate collections");
```

Comment:
This error was bubbled up to the caller previously. I changed it to make the handling consistent with how `create_collections` errors are treated. Lmk if there was a reason for the previous inconsistency, or if we should instead bubble up both errors!

Reply:
this lgtm